### ISSUE TITLE
Another false positive for SC2050 when compared with ""

### ISSUE THREAD

**Issue by dakusui** [koalaman/shellcheck#2653]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2050 
- My shellcheck version (`shellcheck --version` or "online"): 
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

    if [[ "" == "S{_suffix}" ]]; then
      _git_tag_option="--branch ${_image_version}"
    fi

```

#### Here's what shellcheck currently says:

```
This expression is constant. Did you forget the $ on a variable?
See [SC2050](https://github.com/koalaman/shellcheck/wiki/SC2050).
```

#### Here's what I wanted or expected to see:

This shouldn't be reported as it is not a constant.

**Comment by aeiplatform** [comment]
How it is not constant?

**Comment by brother** [comment]
Is it supposed to be an upper case S there or a dollar sign?
`"S{_suffix}"`

Because it surely looks like shellcheck is indeed correct that something is missing a dollar sign or this code is "hard to grasp".
The line inside the if compound does have a dollar sign before the curly braces...
